flaskdeid/annotation.py

```diff
@@ -0,0 +1,110 @@
+"""Module for standardizing and combining annotations"""
+class Annotation(object):
+    def __init__(self, origin):
+        self.origin = origin
+        self.start = None
+        self.end = None
+        self.score = None
+        self.type = None
+        self.text = None
+
+    def empty(self):
+        return not (self.start and self.end and self.text)
+
+    def from_medlp(medlp):
+        ann = Annotation('medlp')
+        ann.start = medlp.get('BeginOffset')
+        ann.end = medlp.get('EndOffset')
+        ann.score = medlp.get('Score')
+        ann.type = medlp.get('Type')
+        ann.text = medlp.get('Text')
+        return ann
+
+    def from_hutchner(hutchner):
+        ann = Annotation('hutchner')
+        ann.start = hutchner.get('start')
+        ann.end = hutchner.get('stop')
+        ann.score = hutchner.get('confidence')
+        ann.type = hutchner.get('label')
+        ann.text = hutchner.get('text')
+        return ann
+
+    def to_dict(self):
+        data = {}
+        data['origin'] = self.origin
+        data['start'] = self.start
+        data['end'] = self.end
+        data['score'] = self.score
+        data['type'] = self.type
+        data['text'] = self.text
+        return data
+
+
+class MergedAnnotation(Annotation):
+    def __init__(self):
+        super().__init__('merged')
+        self.source_annotations = []
+
+    @property
```

Comment:
Clever! Does it complain if there's no setter ?

Reply:
Well, it throws an error (`AttributeError: can't set attribute`), but I think that's actually what we want it to do.